src/coins.cpp

```diff
@@ -107,13 +108,8 @@ void CCoinsViewCache::AddCoin(const COutPoint &outpoint, Coin&& coin, bool possi
 
 void CCoinsViewCache::EmplaceCoinInternalDANGER(COutPoint&& outpoint, Coin&& coin) {
     cachedCoinsUsage += coin.DynamicMemoryUsage();
-    auto [it, inserted] = cacheCoins.emplace(
-        std::piecewise_construct,
```

Comment:
Was this verbose code that is replaced here a workaround for lack of `try_emplace`?

Reply:
It seems this [was introduced in 2017](https://github.com/bitcoin/bitcoin/commit/50830796889ecaa458871f1db878c255dd2554cb#diff-f0ed73d62dae6ca28ebd3045e5fc0d5d02eaaacadb4c2a292985a3fbd7e1c77cR100), when the project was still on [C++11](https://github.com/bitcoin/bitcoin/blob/342b9bc3907edf8eae64440397a32833ed44fae4/configure.ac#L58), but `try_emplace` was only introduced in [C++17](https://en.cppreference.com/w/cpp/container/map/try_emplace).

When this was split out to [`EmplaceCoinInternalDANGER`](https://github.com/bitcoin/bitcoin/commit/f6e2da5fb7c6406c37612c838c998078ea8d2252#diff-095ce1081a930998a10b37358fae5499ac47f8cb6f25f5df5d88e920a54e0341R275), the old impl was copied (on Aug 25, 2020), but C++17 support was only enabled in [Nov 18, 2020](https://github.com/bitcoin/bitcoin/commit/faaee810e62d796d66bfb2bc4e53c8b4c8104d13#diff-3f2b8984fdf0aea4a02d35b54047634465410c1526da78c0b4f888248337d135).

Note that this was already partially upgraded for [FetchCoin](https://github.com/bitcoin/bitcoin/pull/30326/files).